File: src/main/java/io/github/dqualizer/dqlang/types/rqa/definition/stimulus/loadprofile/ConstantLoad.java
package io.github.dqualizer.dqlang.types.rqa.definition.stimulus.loadprofile;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.SuperBuilder;
import io.github.dqualizer.dqlang.types.rqa.definition.stimulus.symbolic.SymbolicValue;

@Getter
@ToString
@SuperBuilder
@NoArgsConstructor
public class ConstantLoad extends LoadProfile {

    /**
     * Type identifier for JSON serialization.
     */
    public static final String TYPE_IDENTIFIER = "constant";

    @JsonProperty("target_load")
    private SymbolicValue targetLoad;
    private SymbolicValue duration;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/rqa/definition/stimulus/loadprofile/ConstantLoad.kt
package io.github.dqualizer.dqlang.types.rqa.definition.stimulus.loadprofile

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.dqualizer.dqlang.types.rqa.definition.stimulus.symbolic.SymbolicValue

data class ConstantLoad(
    @JsonProperty("base_load")
    override val baseLoad: SymbolicValue? = null,

    @JsonProperty("target_load")
    val targetLoad: SymbolicValue? = null,

    val duration: SymbolicValue? = null
) : LoadProfile() {
    companion object {
        const val TYPE_IDENTIFIER: String = "constant"
    }
}
